sdk/eventhubs/azure-eventhubs/src/main/java/com/azure/messaging/eventhubs/EventHubProducer.java

```diff
@@ -198,24 +256,48 @@
         return sendInternal(Flux.from(events), options);
     }
 
+    /**
+     * Sends the batch to the associated Event Hub.
+     *
+     * @param batch The batch to send to the service.
+     * @return A {@link Mono} that completes when the batch is pushed to the service.
+     * @throws NullPointerException if {@code batch} is {@code null}.
+     * @see EventHubProducer#createBatch()
+     * @see EventHubProducer#createBatch(BatchOptions)
+     */
+    public Mono<Void> send(EventDataBatch batch) {
+        Objects.requireNonNull(batch);
+
+        if (batch.getEvents().isEmpty()) {
+            logger.info("Cannot send an EventBatch that is empty.");
+            return Mono.empty();
+        }
+
+        logger.info("Sending batch with partitionKey[{}], size[{}].", batch.getPartitionKey(), batch.getSize());
+
+        final List<Message> messages = EventDataUtil.toAmqpMessage(batch.getPartitionKey(), batch.getEvents());
+
+        return sendLinkMono.flatMap(link -> messages.size() == 1
+            ? link.send(messages.get(0))
+            : link.send(messages));
+    }
+
     private Mono<Void> sendInternal(Flux<EventData> events, SendOptions options) {
         final String partitionKey = options.partitionKey();
 
-        if (!ImplUtils.isNullOrEmpty(partitionKey)) {
-            if (isPartitionSender) {
-                throw new IllegalArgumentException(String.format(Locale.US,
-                    "SendOptions.partitionKey() cannot be set when an EventHubProducer is created with"
-                        + "EventHubProducerOptions.partitionId() set. This EventHubProducer can only send events to partition '%s'.",
-                    senderOptions.partitionId()));
-            } else if (partitionKey.length() > MAX_PARTITION_KEY_LENGTH) {
-                throw new IllegalArgumentException(String.format(Locale.US,
-                    "PartitionKey '%s' exceeds the maximum allowed length: '%s'.", partitionKey, MAX_PARTITION_KEY_LENGTH));
-            }
-        }
+        verifyPartitionKey(partitionKey);
 
-        //TODO (conniey): When we implement partial success, update the maximum number of batches or remove it completely.
         return sendLinkMono.flatMap(link -> {
-            return events.collect(new EventDataCollector(options, 1, link::getErrorContext))
+            //TODO (conniey): When we implement partial success, update the maximum number of batches or remove it completely.
```

Comment:
nit:  is this still relevant?   It's feeling more and more like we're not going to take on auto-splitting for GA.